Server/mods/deathmatch/logic/luadefs/CLuaPedDefs.cpp

```diff
@@ -404,56 +405,33 @@ int CLuaPedDefs::IsPedFrozen(lua_State* luaVM)
     return 1;
 }
 
-int CLuaPedDefs::SetPedAnimation(lua_State* luaVM)
+bool CLuaPedDefs::SetPedAnimation(CElement* pPed, std::optional<std::variant<std::string, std::monostate, bool>> blockName,
+                                  std::optional<std::variant<std::string, std::monostate, bool>> animName, std::optional<int> time, std::optional<bool> loop,
+                                  std::optional<bool> updatePosition, std::optional<bool> interruptable, std::optional<bool> freezeLastFrame,
+                                  std::optional<int> blendTime, std::optional<bool> restoreTask)
 {
-    // bool setPedAnimation ( ped thePed [, string block=nil, string anim=nil, int time=-1, int blend=250, bool loop=true, bool updatePosition=true, bool
-    // interruptable=true, bool freezeLastFrame = true] )
-    CElement* pPed;
-    SString   strBlockName, strAnimName;
-    int       iTime;
-    int       iBlend = 250;
-    bool      bLoop, bUpdatePosition, bInterruptable, bFreezeLastFrame;
-    bool      bDummy;
-    bool      bTaskToBeRestoredOnAnimEnd;
-
-    CScriptArgReader argStream(luaVM);
-    argStream.ReadUserData(pPed);
-    if (argStream.NextIsBool())
-        argStream.ReadBool(bDummy);            // Wiki used setPedAnimation(source,false) as an example
-    else if (argStream.NextIsNil())
-        argStream.m_iIndex++;            // Wiki docs said blockName could be nil
-    else
-        argStream.ReadString(strBlockName, "");
-    argStream.ReadString(strAnimName, "");
-    if (argStream.NextCouldBeNumber())            // Freeroam skips the time arg sometimes
-        argStream.ReadNumber(iTime, -1);
-    else
-        iTime = -1;
-    argStream.ReadBool(bLoop, true);
-    argStream.ReadBool(bUpdatePosition, true);
-    argStream.ReadBool(bInterruptable, true);
-    argStream.ReadBool(bFreezeLastFrame, true);
-    argStream.ReadNumber(iBlend, 250);
-    argStream.ReadBool(bTaskToBeRestoredOnAnimEnd, false);
+    std::string animBlockName = "";
+    std::string animationName = "";
 
-    if (!argStream.HasErrors())
+    if (blockName.has_value())
     {
-        const char *szBlock, *szAnim;
-        szBlock = strBlockName.empty() ? NULL : strBlockName.c_str();
-        szAnim = strAnimName.empty() ? NULL : strAnimName.c_str();
+        if (std::holds_alternative<std::string>(blockName.value()))
```

Comment:
Please, use `value_or`. Otherwise `std::bad_optional_access` exception may be thrown.

Reply:
You mean?
```cpp
if (std::holds_alternative<std::string>(blockName.value_or("")))
```

Reply:
Yes

Reply:
Is this check insufficient? https://github.com/multitheftauto/mtasa-blue/blob/e6dd0c58379ced299afcf29d11b56fc46488e047/Server/mods/deathmatch/logic/luadefs/CLuaPedDefs.cpp#L416

Reply:
Or rather `if (std::holds_alternative<std::string>(blockName.value_or(std::string{})))` because `""` may be considered as a bool.

Reply:
> Is this check insufficient?
> 
> https://github.com/multitheftauto/mtasa-blue/blob/e6dd0c58379ced299afcf29d11b56fc46488e047/Server/mods/deathmatch/logic/luadefs/CLuaPedDefs.cpp#L416

Oh, yeah. Sorry, I didn't notice that condition.